src/controller/CHIPCluster.h

```diff
@@ -58,6 +59,7 @@ class DLL_EXPORT ClusterBase
     CHIP_ERROR AssociateWithGroup(DeviceProxy * device, GroupId groupId);
 
     void Dissociate();
+    void SetCommandTimeout(Optional<System::Clock::Timeout> timeout) { mTimeout = timeout; }
```

Comment:
Worthwhile adding a comment/TODO here indicating that this is temporary while the CHIPClusters.cpp code transitions over to using `InvokeCommand` below.

Since you've done most of the leg-work already, I'd recommend patching up InvokeCommand below as well to take a timeout arg, which will setup well for later.

Reply:
How would the new command path work from the  higher levels? Right now, the higher-level code calls associate then set command timeout. Higher level code currently calls the generated command functions from the derived cluster classes, which don't have a space for the timeout. The timeout that was set in the base class is added by the derived clusters when they calls SendCommands. Is the plan to have the upper levels call InvokeCommand rather than going through the generated code?

Reply:
> Is the plan to have the upper levels call InvokeCommand rather than going through the generated code?

See https://github.com/project-chip/connectedhomeip/pull/13371

Reply:
Ah OK. Also sad because that's going to definitely blow up my other PR, that's not in because it got blown up by another change that went in in dec. :(

Regardless, I'll add the todo.